File: rebus-rebuscrus/src/java/it/csi/rebus/rebuscrus/vo/DepositoVO.java
/*
* SPDX-FileCopyrightText: (C) Copyright 2022 Regione Piemonte
* SPDX-License-Identifier: EUPL-1.2
*/
package it.csi.rebus.rebuscrus.vo;

@SuppressWarnings("serial")
public class DepositoVO extends ParentVO implements Comparable<DepositoVO> {
	private Long id;
	private String denominazione;
	private Boolean isPrevalente;
	private String telefono;
	private String indirizzo;

	public Long getId() {
		return id;
	}

	public void setId(Long id) {
		this.id = id;
	}

	public String getDenominazione() {
		return denominazione;
	}

	public void setDenominazione(String denominazione) {
		this.denominazione = denominazione;
	}

	public Boolean getIsPrevalente() {
		return isPrevalente;
	}

	public void setIsPrevalente(Boolean isPrevalente) {
		this.isPrevalente = isPrevalente;
	}

	public String getTelefono() {
		return telefono;
	}

	public void setTelefono(String telefono) {
		this.telefono = telefono;
	}

	public String getIndirizzo() {
		return indirizzo;
	}

	public void setIndirizzo(String indirizzo) {
		this.indirizzo = indirizzo;
	}

	@Override
	public int compareTo(DepositoVO o) {
		return this.denominazione.compareTo(o.getDenominazione());
	}
}

File: rebus-rebuscruswcl/src/app/vo/deposito-vo.ts
/*************************************************
Copyright Regione Piemonte - 2022
SPDX-License-Identifier: EUPL-1.2-or-later
***************************************************/

export class DepositoVO {
    id: number;
    denominazione: string;
    isPrevalente: boolean;
    telefono: string;
    indirizzo: string;

    constructor(depositoVO: DepositoVO) {
        this.id = depositoVO.id;
        this.denominazione = depositoVO.denominazione;
        this.isPrevalente = depositoVO.isPrevalente;
        this.telefono = depositoVO.telefono;
        this.indirizzo = depositoVO.indirizzo;
    }
}